core/templates/domain/admin/job.model.ts

```diff
@@ -37,17 +37,17 @@ export class Job {
   timeFinishedMsec: number;
   jobType: string;
   statusCode: string;
-  error: string;
-  canBeCanceled: boolean;
+  error: string | null;
+  canBeCanceled: boolean | undefined;
```

Comment:
Why making a boolean undefined ? Can we make this property to be not optional as when this is undefined it will be false only?

Reply:
we are actually getting the data through http requests and the JSON object being returned for `recentJobData` would not have this property. so, even if we keep the type as `boolean`, the value will still be `undefined` when we get the object.

Reply:
Not really sure what you meant here, maybe mentors can help @srijanreddy98 @vojtechjelinek PTAL

Reply:
@brianrodri has a better understanding of this... Is there a chance that `canBeCanceled` be undefined?

Reply:
I think the response could have it missing, but we should keep it as a boolean here (if it _is_ missing, then just make it false)

Reply:
done! it is `false` wherever it is missing